src/llm_models/llm_model.rs

```diff
@@ -69,4 +70,16 @@ pub trait LLMModel {
         //To be safe we go with smaller of the numbers
         std::cmp::min(max_requests_from_rpm, max_requests_from_tpm)
     }
+    ///Returns the default temperature to be used by the model
+    fn get_default_temperature(&self) -> f32 {
+        0f32
```

Comment:
default to 0 unless otherwise specified by the model

Reply:
```ChatGPT generates text through a probabilistic language modeling approach, where it writes responses word by word, calculating the most likely next word in the sequence. A key feature that influences this behavior is called temperature [[8](https://pmc.ncbi.nlm.nih.gov/articles/PMC10960206/#ref8), [9](https://pmc.ncbi.nlm.nih.gov/articles/PMC10960206/#ref9)]. In this context, temperature is a value from 0 to 2 that adjusts how random each subsequent word in the chat output is. A value of 0 will give the most probable word and, thus, the least variability. As the value increases toward and beyond 1, the next word becomes less probable, leading to more randomness and “creativity” in the response```

Reply:
Correct. In the context of the crate, when we are using LLMs for generating type-safe responses, the default of more predictable and less creative seems reasonable, right?

Reply:
yes